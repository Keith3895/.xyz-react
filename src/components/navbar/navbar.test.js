import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import NavBar from './navbar';
import pretty from "pretty";
import { BrowserRouter as Router } from 'react-router-dom';
let container = null;
beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

describe('Hamburger icon', () => {
    it('Is hamburger rendered', () => {
        act(() => {
            render(
                <Router>
                    <NavBar />
                </Router>
                , container);
        });
        let a = container.querySelector('.cd-3d-nav-trigger span');
        expect(a).toBeTruthy();
        let navbar = container.querySelector('.cd-3d-nav-container.nav-is-visible');
        expect(navbar).toBeFalsy();
    });
    it('on cilck of hamburger', () => {
        act(() => {
            render(
                <Router>
                    <NavBar active={()=>{}}/>
                </Router>
                , container);
        });
        let a = container.querySelector('.cd-3d-nav-trigger');
        a.click();
        let span = container.querySelector('.nav-is-visible .cd-3d-nav-trigger span');
        expect(span).toBeTruthy();
        let navbar = container.querySelector('.cd-3d-nav-container.nav-is-visible');
        expect(navbar).toBeTruthy();
    });
    
    it('on cilck of X', () => {
        act(() => {
            render(
                <Router>
                    <NavBar active={()=>{}}/>
                </Router>
                , container);
        });
        let a = container.querySelector('.cd-3d-nav-trigger');
        a.click();
        let span = container.querySelector('.nav-is-visible .cd-3d-nav-trigger span');
        expect(span).toBeTruthy();
        let navbar = container.querySelector('.cd-3d-nav-container.nav-is-visible');
        expect(navbar).toBeTruthy();
        a.click();
        span = container.querySelector('.nav-is-visible .cd-3d-nav-trigger span');
        expect(span).toBeFalsy();
        navbar = container.querySelector('.cd-3d-nav-container.nav-is-visible');
        expect(navbar).toBeFalsy();
    });
    it('render all item.', () => {
        act(() => {
            render(
                <Router>
                    <NavBar active={()=>{}}/>
                </Router>
                , container);
        });
        let a = container.querySelector('.cd-3d-nav-trigger');
        a.click();
        let span = container.querySelector('.nav-is-visible .cd-3d-nav-trigger span');
        expect(span).toBeTruthy();
        let navbar = container.querySelector('.cd-3d-nav-container.nav-is-visible');
        expect(navbar).toBeTruthy();
        let list =  container.querySelectorAll('li');
        expect(list.length).toBe(4);
    });
    it('onclick line item of navbar',()=>{
        act(() => {
            render(
                <Router>
                    <NavBar active={()=>{}}/>
                </Router>
                , container);
        });
        let a = container.querySelector('.cd-3d-nav-trigger');
        a.click();
        let span = container.querySelector('.nav-is-visible .cd-3d-nav-trigger span');
        expect(span).toBeTruthy();
        let marker = container.querySelector('.cd-marker.color-1');
        let active = container.querySelector('.el.cd-selected');
        let list =  container.querySelectorAll('li');
        list[1].firstElementChild.click();
        expect(list[1].firstElementChild.className).toBe('el cd-selected');
        span = container.querySelector('.nav-is-visible .cd-3d-nav-trigger span');
        expect(span).toBeFalsy();
    });
    it('snap shot test',()=>{
        act(() => {
            render(
                <Router>
                    <NavBar active={()=>{}}/>
                </Router>
                , container);
        });
        expect(pretty(container.innerHTML)).toMatchSnapshot();
    });
});
