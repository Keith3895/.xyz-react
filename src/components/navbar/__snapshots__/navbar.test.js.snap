// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger icon snap shot test 1`] = `
"<div>
  <header class=\\"cd-header \\"><a class=\\"cd-3d-nav-trigger\\">Menu<span></span></a></header>
  <nav class=\\"cd-3d-nav-container \\">
    <ul class=\\"cd-3d-nav\\">
      <li><a class=\\"el\\" href=\\"/\\">Home</a></li>
      <li><a aria-current=\\"page\\" class=\\"el cd-selected\\" href=\\"/project\\">Project</a></li>
      <li><a class=\\"el\\" href=\\"/blog\\">Blogs</a></li>
      <li><a class=\\"el\\" href=\\"/contact\\">Contact</a></li>
    </ul><span class=\\"cd-marker color-1\\" style=\\"left: 0px;\\"></span>
  </nav>
</div>"
`;
